### ISSUE TITLE
Re-Open: Filtered sync with _deleted not working with put

### ISSUE THREAD

**Issue by ddolheguy** [pouchdb/pouchdb#5912]
I'd like to re-open this issue (https://github.com/pouchdb/pouchdb/issues/5906) as I believe this might be broken in the current build?

All comments have been added to the above issue to keep it in context and there is a lot of details around the issue.  I do not have access to Re-Open issues as requested so I'm using this as a request to re-open instead.

**Comment by nolanlawson** [comment]
I read through your comments. Can you set a breakpoint on your filter function and see if your document is passing the filter function or not? You can add the line `debugger;` to accomplish this.

Also: if you remove the filter function, does the deleted document replicate? That is a good way to figure out if the error is due to the filter function or to something else.

**Comment by ddolheguy** [comment]
Ok, before getting to the delete issue, I wanted to start from scratch and make sure everything is working.  I found first that when I updated (edit) a document in PouchDB, whilst sync'ing to CouchDB (2.0), I get an instant "conflict" in CouchDB?

Steps to reproduce:
1. Start a sync in PouchDB to CouchDB (2.0) using the below config:

```
this.db.sync('http://localhost:5984/mydatabase', {
   live: true,
   retry: true
});
```

2. Create a document in PouchDB
2. Make sure it syncs to CouchDB 
3. Update the document in PouchDB using the following process:

```
database.get(id)
   .then(doc => {
       doc.someData = 'Changed'
       return database.put(doc);
    })
```

4. Upon put, the sync in PouchDB works and "pushes" the document to CouchDB, but then I get a "pull" from CouchDB of the same document with but with the previous revision?

Looking in CouchDB, I see the updated document (the winner is the correct version), but there is now a "Conflict" tab which shows a conflict?

If you update again on PouchDB, the same thing happens and you get a third conflict?  Why is this happening?

PS: I'm thinking this might be a solution to the above ```_delete: true ``` issue since that is also an update.
